tests/infra/test_with_custom_state.py

```diff
@@ -0,0 +1,93 @@
+import pytest
+
+from eth2spec.test.context import (
+    with_custom_state,
+    default_activation_threshold,
+    default_balances,
+    zero_activation_threshold,
+)
+from eth2spec.test.helpers.constants import MINIMAL, PHASE0
+from eth2spec.test.helpers.specs import spec_targets
+
+
+class TestWithCustomState:
+    """Test suite for with_custom_state decorator."""
+
+    def test_with_custom_state_injects_state_view(self):
+        """Test that the decorator injects a BeaconState with expected properties."""
+        spec = spec_targets[MINIMAL][PHASE0]
+
+        @with_custom_state(default_balances, default_activation_threshold)
+        def test_case(*, spec, phases, state):
+            # Verify the state is properly initialized
+            assert len(state.validators) > 0
```

Comment:
@DeborahOlaboye this is not checking that the state is as it should be. What are the balcnes and the activated ones? Are the ones that should come from this?